### ISSUE TITLE
Add systemd native logging support

### ISSUE THREAD

**Issue by johnsonjh** [aremmell/libsir#133]
See http://0pointer.de/public/systemd-man/sd_journal_print.html

This is surprisingly simple and using syslog levels that map essentially directly to SIRs defined levels (LOG_EMERG, LOG_ALERT, LOG_CRIT, LOG_ERR, LOG_WARNING, LOG_NOTICE, LOG_INFO, LOG_DEBUG).  It would be essentially no different than the existing syslog code, but gain some automatic benefits:

> [syslog(3)](http://man7.org/linux/man-pages/man3/syslog.3.html) and sd_journal_print() may largely be used interchangeably functionality-wise. However, note that log messages logged via the former take a different path to the journal server than the later, and hence global chronological ordering between the two streams cannot be guaranteed. Using sd_journal_print() has the benefit of logging source code line, filenames, and functions as metadata along all entries, and guaranteeing chronological ordering with structured log entries that are generated via sd_journal_send().

Since it would add an explicit dependency on libsystemd, I'd recommend that the addition to CFLAGS and LDFLAGS (`$(pkg-config --cflags --libs libsystemd)`) be an option, like "SIR_OS_LOGGER_SYSTEMD=1".  The other option would be to try to find and dynamically load libsystemd at runtime, and then use it if that was successful.  This might be a good approach too.

Not high priority - I'll do the work.

**Comment by johnsonjh** [comment]
Actually, it would make the most sense to just use `sd_journal_send` so the logged file/function/line could be properly passed from libsir.  The fields to use are documented at http://0pointer.de/public/systemd-man/systemd.journal-fields.html

**Comment by aremmell** [comment]
So I read the docs at that URL, and I saw:

> syslog(3) and sd_journal_print() may largely be used interchangeably functionality-wise.

Sounds like we already write to the systemd journal via syslog?

**Comment by johnsonjh** [comment]
Not necessarily.  The systemd's journal facility supports the syslog API.  Also, syslog daemons have journal integrations, but they're totally different things.  You can run a system without a syslog daemon at all, only journald, or you can run a system without journald and just syslog, and you can run a system with both syslog and journald - they aren't necessarily connected.

**Comment by johnsonjh** [comment]
For example, I normally use rsyslog on all my machines, which supports the old-style syslog and the new-style journal.

I can configure rsyslog to work with the journal and with elasticsearch, sending everything to elasticsearch (or splunk, or whatever), which is very common - see https://sematext.com/blog/journald-logging-tutorial/#journald-vs-syslog for example.

Syslog doesn't support much of anything, you just get the time, a priority, a "facility", etc.

But with the journal, you get dozens and dozens of fields - https://www.freedesktop.org/software/systemd/man/systemd.journal-fields.html and you can also set your own.

Like, here's what a journal entry looks like logged via the journal method:

```json
{
  "CODE_LINE": "768",
  "__REALTIME_TIMESTAMP": "1695724402719502",
  "_SYSTEMD_UNIT": "init.scope",
  "JOB_TYPE": "start",
  "_CMDLINE": "/usr/lib/systemd/systemd --system --deserialize=27",
  "CODE_FUNC": "job_emit_done_message",
  "TID": "1",
  "JOB_ID": "3236963",
  "CODE_FILE": "src/core/job.c",
  "_UID": "0",
  "__MONOTONIC_TIMESTAMP": "4589724420257",
  "_PID": "1",
  "PRIORITY": "6",
  "__CURSOR": "s=0a2a889f461e4fa18954c56dd8b8c62c;i=1435fb8;b=e00e35fc864e49f480d6490fe2354a5c;m=42ca0e4c8a1;t=606409aa93b0e;x=24610ef1b24eeb6",
  "_SYSTEMD_CGROUP": "/init.scope",
  "SYSLOG_IDENTIFIER": "systemd",
  "MESSAGE_ID": "39f53479d3a045ac8e11786248231fbf",
  "SYSLOG_FACILITY": "3",
  "MESSAGE": "Started docker-9e602e567cb0ecb16d7809b53e0e8cb32f52697e4313d55acdafe1ef024eddeb.scope - libcrun container.",
  "_SYSTEMD_SLICE": "-.slice",
  "_HOSTNAME": "brshpc0",
  "UNIT": "docker-9e602e567cb0ecb16d7809b53e0e8cb32f52697e4313d55acdafe1ef024eddeb.scope",
  "_GID": "0",
  "_MACHINE_ID": "a3f6faac3e2646f49a33636bf7a1bb34",
  "_SOURCE_REALTIME_TIMESTAMP": "1695724402719462",
  "INVOCATION_ID": "fa5cc0624d25438ca1134d12e7513221",
  "JOB_RESULT": "done",
  "_COMM": "systemd",
  "_BOOT_ID": "e00e35fc864e49f480d6490fe2354a5c",
  "_EXE": "/usr/lib/systemd/systemd",
  "_TRANSPORT": "journal",
  "_CAP_EFFECTIVE": "1ffffffffff",
  "_RUNTIME_SCOPE": "system"
}
```

And here's an entry in the journal that came in via syslog:

```json
{      
  "__CURSOR": "s=0a2a889f461e4fa18954c56dd8b8c62c;i=1436239;b=e00e35fc864e49f480d6490fe2354a5c;m=42ca55751ec;t=606409f1bc459;x=3dbe31092e2e28cf",                                                                                 
  "__MONOTONIC_TIMESTAMP": "4589799035372",
  "_SYSTEMD_INVOCATION_ID": "8f3b35ef22ef4778b539f90c6215ac3b",                                                  
  "_COMM": "sshd",
  "_SOURCE_REALTIME_TIMESTAMP": "1695724477334580",
  "_TRANSPORT": "syslog",
  "__REALTIME_TIMESTAMP": "1695724477334617",
  "SYSLOG_TIMESTAMP": "Sep 26 06:34:37 ",
  "PRIORITY": "2",
  "MESSAGE": "fatal: Timeout before authentication for 144.34.212.238 port 53888",                               
  "_HOSTNAME": "brshpc0",
  "_SYSTEMD_CGROUP": "/system.slice/sshd.service",
  "_BOOT_ID": "e00e35fc864e49f480d6490fe2354a5c",
  "_GID": "0",
  "SYSLOG_IDENTIFIER": "sshd",
  "_SYSTEMD_UNIT": "sshd.service",
  "_UID": "0",
  "_PID": "3456472",
  "SYSLOG_PID": "3456472",
  "_CAP_EFFECTIVE": "1ffffffffff",
  "_MACHINE_ID": "a3f6faac3e2646f49a33636bf7a1bb34",
  "_RUNTIME_SCOPE": "system",
  "_SYSTEMD_SLICE": "system.slice",
  "SYSLOG_FACILITY": "4"
}      
```

**Comment by johnsonjh** [comment]
Also see #121
